runtime/doc/lua.txt

```diff
@@ -34,20 +34,32 @@ Nvim ever ships with Lua 5.4+, a Lua 5.1 compatibility shim will be provided
 so that old plugins continue to work transparently.
 
                                                                   *lua-luajit*
-Nvim is built with luajit on platforms which support it, which provides
-extra functionality. Lua code in |init.lua| and plugins can assume its presence
-on installations on common platforms. For maximum compatibility with less
-common platforms, availability can be checked using the `jit` global variable: >lua
+On supported platforms, Nvim is built with LuaJIT, which provides extra
+functionality (compared to PUC Lua) such as "bit" and various utilities in
+`$VIMRUNTIME/lua/jit/`. Lua code in |init.lua| and plugins can assume its
+presence on many platforms, but for maximum compatibility should check the
+`jit` global variable: >lua
   if jit then
     -- code for luajit
   else
     -- code for plain lua 5.1
   end
 <
                                                                   *lua-bit*
-In particular, the luajit "bit" extension module is _always_ available.
-A fallback implementation is included when nvim is built with PUC Lua 5.1,
-and will be transparently used when `require("bit")` is invoked.
+The LuaJIT "bit" extension module is _always_ available: when built with PUC
+Lua, Nvim includes a fallback implementation which provides `require("bit")`.
+
+                                                                  *lua-profile*
+To profile Lua code (with LuaJIT-enabled Nvim), the basic steps are: >lua
+    -- Start a profiling session:
+    require('jit.p').start('i1', '/tmp/profile')
```

Comment:
Should we encourage users to use the `r` option to show absolute timing rather than the percentages in the example? Though we probably don't want to explain here verbosely about what's available in `jit.p`. 

Ref https://github.com/neovim/neovim/pull/28723#issuecomment-2106637108

We can also consider adding links to external doc, e.g. https://luajit.org/ext_profiler.html